Generator/Generate.ts

```diff
@@ -0,0 +1,113 @@
+import {Board} from './Board';
+import { StrategyEnum } from './Sudoku';
+
+const events = require('events');
+const fs = require('fs');
+const readline = require('readline');
+const filepath:String = process.argv[2];
+
+/**
+ * Returns the user specified start index if there was one provided, otherwise returns 1
+ * @returns puzzle index to start generating (1 indexed, inclusive)
+ */
+function getStart():number {
+    if (process.argv[3] === undefined) {
+        return 1;
+    }
+    return Number(process.argv[3]);
+}
+
+/**
+ * Returns the user specified end index if there was one provided, otherwise returns one billion
+ * @returns index of last puzzle to generate (1 indexed, inclusive)
+ */
+function getEnd():number {
+    if (process.argv[4] === undefined) {
+        return 1_000_000_000;
+    }
+    return Number(process.argv[4]);
+}
+
+/**
+ * Returns the user specified batch size if there was one provided, otherwise returns one billion
+ * @returns batch size which determines how many puzzles per array (one array per line)
+ */
+function getBatchSize():number {
+    if (process.argv[5] === undefined) {
+        return 1_000_000_000;
```

Comment:
Could we change default maxBatchSize to 1,000 instead of 1 billion?
https://stackoverflow.com/questions/36042967/mongoose-insertmany-limit#:~:text=I%27d%20recommend%20based%20on%20personal%20experience%2C%20batch%20of,group%20into%20smaller%20groups%20of%201000%20or%20less.
https://www.mongodb.com/docs/manual/reference/method/db.collection.insertMany/#execution-of-operations